### DISCUSSION TITLE
Cache

### DISCUSSION THREAD

**Discussion by VirtualSim** [pelicanmapping/osgearth#2089] [Support Q&A]
Hi there,

First, I am impressed with version 3.2 compared to 2.8, regarding the rendering in 3D and the combining of OpenFlt terrain incrustation.

Now, I found that .earth files have changed a lot and could not get the new stuff in the forum or documentation.

So: How now to enable cache (and directory) to avoid having the blurred earth and reload again and again from internet.

Also, Sky seems not created (osgEarth::findTopMostNodeOfType<SkyNode>(root); return null). I can create it but shouldn't it be created at load() with some options in the .earth file ?

Last, I follow a satellite with the earth below. How to force the tiles to be of higher resolution and not blurred ?

Thanks.

**Comment by GISlxz** [comment]
"Regarding the cache options for the new version of osgEarth, it seems that the registry's setDefaultCache function should be used. However, after briefly reviewing the cacheOptions and cacheFactory files, I haven't quite figured out how to create a cache in a specific directory. I have an alternative approach, which is to specify the cache directory via an environment variable: _putenv_s("OSGEARTH_CACHE_PATH", "D:/coding/testdata/cache");."
